### PR TITLE
Reconstruct failed actors without sending tasks.

### PR DESCRIPTION
**Pull request by raulchen**
## What do these changes do?

Previously, we had to send a task to trigger the reconstruction of a failed actor. This has issues in some cases. For example, an actor that reading data from external DB will never receive tasks. This PR fixes this issue.

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Review by zhijunfu — APPROVED** [review]
looks good to me. thanks

File: python/ray/tests/utils.py [hunk 1]
```diff
@@ -94,3 +94,25 @@ def wait_for_errors(error_type, num_errors, timeout=10):
             return
         time.sleep(0.1)
     raise Exception("Timing out of wait.")
+
+
+def wait_for_contition(condition_predictor,
```
Comment (stephanie-wang) [inline python/ray/tests/utils.py]:
```suggestion
def wait_for_condition(condition_predictor,
```

File: python/ray/tests/utils.py [hunk 2]
```diff
@@ -94,3 +94,25 @@ def wait_for_errors(error_type, num_errors, timeout=10):
...
+                       timeout_ms=1000,
+                       retry_interval_ms=100):
+    """A helper function that wait until a conition is met.
```
Comment (stephanie-wang) [inline python/ray/tests/utils.py]:
```suggestion
    """A helper function that waits until a condition is met.
```

**Review by stephanie-wang — CHANGES_REQUESTED** [review]
Thanks, this looks good!

**Comment by stephanie-wang** [comment]
Looks like the unit test that was added failed on one of the Travis runs: https://travis-ci.com/ray-project/ray/jobs/215417720. We should increase the timeout for that test.

**Comment by raulchen** [comment]
thanks, increased to 5s

**Comment by raulchen** [comment]
@stephanie-wang Tests have passed. Can you give a stamp?

**Review by stephanie-wang — APPROVED** [review]
Thanks! :)

**Pull request merged**